troi/webserver/main.py

```diff
@@ -32,19 +32,19 @@ def page_not_found(e):
 
 def error_check_arguments(inputs):
 
-    args = []
+
+    args = {}
     for input in inputs:
         arg = request.args.get(input['name'], '')
         if not arg and not input['optional']:
             return "Parameter %s is missing." % input['name'], ()
         try:
-            args.append(input['type'](arg))
+            args[input['name']] = input['type'](arg)
         except:
             if input['optional']:
                 args.append(None)
```

Comment:
Since `args` is now a `dict`, we'll need to change this accordingly.

Reply:
Good catch!